packages/dev/core/src/AudioV2/abstractAudio/subProperties/spatialAudioListener.ts

```diff
@@ -1,20 +1,134 @@
+import type { TransformNode } from "../../../Meshes/transformNode";
+import type { Nullable } from "../../../types";
+import { _SpatialAudioAttacher } from "../spatialAttachers/spatialAudioAttacher";
+import type { SpatialAudioAttachmentType } from "../spatialAttachers/spatialAudioAttacher";
 import type { ISpatialAudioListenerOptions } from "./abstractSpatialAudioListener";
 import { _SpatialAudioListenerDefaults, AbstractSpatialAudioListener } from "./abstractSpatialAudioListener";
 
 /** @internal */
 export abstract class _SpatialAudioListener extends AbstractSpatialAudioListener {
+    private _attacher: _SpatialAudioAttacher;
+
+    protected constructor() {
+        super();
+
+        this._attacher = new _SpatialAudioAttacher(this);
+    }
+
+    /** @internal */
+    public get attachedCamera(): Nullable<Camera> {
+        return this._attacher.attachedCamera;
+    }
+
+    public set attachedCamera(value: Nullable<Camera>) {
+        this._attacher.attachedCamera = value;
+    }
+
     /** @internal */
-    public setOptions(options: Partial<ISpatialAudioListenerOptions>): void {
-        if (options.listenerPosition !== undefined) {
+    public get attachedMesh(): Nullable<AbstractMesh> {
+        return this._attacher.attachedMesh;
+    }
+
+    /** @internal */
+    public set attachedMesh(value: Nullable<AbstractMesh>) {
+        this._attacher.attachedMesh = value;
+    }
+
+    /** @internal */
+    public get attachedTransformNode(): Nullable<TransformNode> {
+        return this._attacher.attachedTransformNode;
+    }
+
+    /** @internal */
+    public set attachedTransformNode(value: Nullable<TransformNode>) {
+        this._attacher.attachedTransformNode = value;
+    }
+
+    /** @internal */
+    public get attachmentType(): SpatialAudioAttachmentType {
+        return this._attacher.attachmentType;
+    }
+
+    public set attachmentType(value: SpatialAudioAttachmentType) {
+        this._attacher.attachmentType = value;
+    }
+
+    /** @internal */
+    public get isAttached(): boolean {
+        return this._attacher.isAttached;
+    }
+
+    /** @internal */
+    public get isAttachedToPosition(): boolean {
+        return this._attacher.isAttachedToPosition;
+    }
+
+    /** @internal */
+    public get isAttachedToRotation(): boolean {
+        return this._attacher.isAttachedToRotation;
+    }
+
+    /** @internal */
+    public get minUpdateTime(): number {
+        return this._attacher.minUpdateTime;
+    }
+
+    public set minUpdateTime(value: number) {
+        this._attacher.minUpdateTime = value;
+    }
+
+    /**
+     * Detaches the audio source from the currently attached camera, mesh or transform node.
+     */
+    public detach(): void {
+        this._attacher.detach();
+    }
+
+    /** @internal */
+    public dispose(): void {
+        this._attacher.dispose();
+    }
+
+    /** @internal */
+    public async setOptions(options: Partial<ISpatialAudioListenerOptions>): Promise<void> {
```

Comment:
A comment made by Seb - why is this async now? there is nothing async in the function body (unless I missed something?). Or is it due to inheritance from a different branch?

The same goes to the other setOptions function changed in this PR.

Reply:
They are both async because they await `this._attacher.isReadyPromise;`.